xilem/src/view/flex.rs

```diff
@@ -131,69 +131,579 @@ where
-            .expect("ElementSplice::mutate won't overflow");
+    fn mutate<R>(&mut self, f: impl FnOnce(Mut<'_, FlexElement>) -> R) -> R {
+        let child = FlexElementMut {
+            parent: self.element.reborrow_mut(),
+            idx: self.idx,
+        };
         let ret = f(child);
-        // Skip past the implicit spacer as well as this child
-        self.ix += 2;
+        self.idx += 1;
         ret
     }
 
-    fn delete<R>(
-        &mut self,
-        f: impl FnOnce(<Pod<Box<dyn Widget>> as xilem_core::ViewElement>::Mut<'_>) -> R,
-    ) -> R {
-        let child = self
-            .element
-            .child_mut(self.ix)
-            .expect("ElementSplice::mutate won't overflow");
-        let ret = f(child);
-        self.element.remove_child(self.ix);
-        // Also remove the implicit spacer
-        // TODO: Make the spacers be explicit?
-        self.element.remove_child(self.ix);
-
+    fn delete<R>(&mut self, f: impl FnOnce(Mut<'_, FlexElement>) -> R) -> R {
+        let ret = {
+            let child = FlexElementMut {
+                parent: self.element.reborrow_mut(),
+                idx: self.idx,
+            };
+            f(child)
+        };
+        self.element.remove_child(self.idx);
         ret
     }
 
     fn skip(&mut self, n: usize) {
-        self.ix += n * 2;
+        self.idx += n;
+    }
+}
+
+/// A trait which extends a [`WidgetView`] with methods to provide parameters for a flex item, or being able to use it interchangably with a spacer
+pub trait FlexExt<State, Action>: WidgetView<State, Action> {
+    /// Applies [`impl Into<FlexParams>`](`FlexParams`) to this view, can be used as child of a [`Flex`] [`View`]
+    ///
+    /// # Examples
+    /// ```
+    /// use xilem::{view::{button, label, flex, CrossAxisAlignment, FlexSpacer, FlexExt}};
+    /// # use xilem::{WidgetView};
+    ///
+    /// # fn view<State: 'static>() -> impl WidgetView<State> {
+    /// flex((
+    ///     button("click me", |_| ()).flex(2.0),
+    ///     FlexSpacer::Fixed(2.0),
+    ///     label("a label").flex(CrossAxisAlignment::Fill),
+    ///     FlexSpacer::Fixed(2.0),
+    /// ))
+    /// # }
+    ///
+    /// ```
+    fn flex(self, params: impl Into<FlexParams>) -> FlexItem<Self, State, Action>
+    where
+        State: 'static,
+        Action: 'static,
+        Self: Sized,
+    {
+        flex_item(self, params)
+    }
+
+    /// Turns this [`WidgetView`] into an [`AnyFlexChild`],
+    /// which can be used interchangably with an `FlexSpacer`, as child of a [`Flex`] [`View`]
+    ///
+    /// # Examples
+    /// ```
+    /// use xilem::{view::{flex, label, FlexSpacer, FlexExt, AnyFlexChild}};
+    /// # use xilem::{WidgetView};
+    ///
+    /// # fn view<State: 'static>() -> impl WidgetView<State> {
+    /// flex([label("a label").into_any_flex(), AnyFlexChild::Spacer(FlexSpacer::Fixed(1.0))])
+    /// # }
+    ///
+    /// ```
+    fn into_any_flex(self) -> AnyFlexChild<State, Action>
```

Comment:
I've tried to do this via a blanket impl, but it doesn't work, since I can't specialize over `View<Element = FlexElement>` and `WidgetView`, and otherwise that impl intersects, so I hardcoded these directly on either `V: WidgetView` as well as for either `FlexItem` and `FlexSpacer`...

Reply:
Did you try `View<Element = Pod<W>>` and `View<Element = FlexElement>`? The first should be equivalent to `WidgetView`, but might be more friendly to the compiler

Reply:
Yep I tried, I did not make it to work, unfortunately...